### PR TITLE
added solar integration

### PR DESCRIPTION
**Pull request by jgrayson676**

### CODE REVIEW

File: src/common.coffee [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# humanize known error responses
+cleanmsg = (value) ->
+  if value == 'Provider lookup failed Parameter name: Provider'
+    value = 'Provider lookup failed'
+  else if value == 'Client lookup failed Parameter name: Client'
+    value = 'Missing or invalid client key'
+  else if value == 'Client was not specified Parameter name: Client'
+    value = 'Missing client parameter'
+# this only gets returned if NO lead data is passed.  Passing any singular field results in success
```
Comment (cgrayson) [inline src/common.coffee]: Minor: fix indentation

File: src/solar.coffee [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    postal_code: vars.lead.postal_code
+    phone_number: vars.lead.phone_1
+    alt_phone: vars.lead.phone_2
+    email: vars.lead.email
+
+    utility_company: vars.lead.utility?.electric.company?.name
+    low_bill: vars.lead.utility?.electric.monthly_amount?.min
+    high_bill: vars.lead.utility?.electric.monthly_amount?.max
+    form_comments: vars.lead.comments
+    prime_source: vars.prime_source
```
Comment (cgrayson) [inline src/solar.coffee]: I don't recognize this. Should it be `vars.lead.prime_source`?

File: src/solar.coffee [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  for key, value of content
+    content[key] = value?.valueOf()
+    delete content[key] unless value
+  for key, value of flatten(vars.velocify?.custom ? {})
+    content[key] = value?.valueOf()
+
+  # URL encoded post body
+  content = querystring.stringify(content)
+
+  url: "https://secure.velocify.com/Import.aspx?Provider=LeadConduitSolar&Client=#{vars.velocify?.client}&CampaignId=#{vars.velocify?.campaign_id}&XmlResponse=true"
```
Comment (cgrayson) [inline src/solar.coffee]: I see there's no `validate()` function in this module (I think it was just written before support for `validate()` existed). While you're in here, I think go ahead and add it, for both `mortgage` and `solar`. Seems like without this `client` and `campaign_id`, at least, the request would be doomed to fail.

File: src/solar.coffee [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  body: content
+
+
+request.variables = ->
+  [
+    { name: 'velocify.client', type: 'string', required: true, description: 'Client key for Velocify account'}
+    { name: 'velocify.campaign_id', type: 'string', required: true, description: 'The ID of the Velocify campaign'}
+    { name: 'lead.first_name', type: 'string', description: 'First name'}
+    { name: 'lead.last_name', type: 'string', description: 'Last name'}
+    { name: 'lead.address_1', type: 'string', description: 'Street address'}
```
Comment (cgrayson) [inline src/solar.coffee]: Missing `lead.email`; double check this array matches `request()`.

File: src/solar.coffee [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    { name: 'outcome', type: 'string', description: 'The outcome of the transaction (default is success)'}
+    { name: 'reason', type: 'string', description: 'If the outcome was a failure, this is the reason'}
+  ]
+
+#
+# Exports ----------------------------------------------------------------
+#
+
+module.exports =
+  name: 'solar',
```
Comment (cgrayson) [inline src/solar.coffee]: This is what shows in the UI; should be capitalized.

File: src/solar.coffee (hunk 4 above)
Comment (cgrayson) [inline src/solar.coffee]: Assuming none of the rest of these are required, let's go ahead and have `required: false` on 'em.

**Comment by cgrayson** [comment]
Also: noticed the `license` field in `package.json` is the old/wrong format, if you don't updating that.

Let me know when you're ready to have this looked over again.

File: src/solar.coffee [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    postal_code: vars.lead.postal_code
+    phone_number: vars.lead.phone_1
+    alt_phone: vars.lead.phone_2
+    email: vars.lead.email
+
+    utility_company: vars.lead.utility?.electric.company?.name
+    low_bill: vars.lead.utility?.electric.monthly_amount?.min
+    high_bill: vars.lead.utility?.electric.monthly_amount?.max
+    form_comments: vars.lead.comments
+    prime_source: vars.velocify.prime_source or 'paid leads'
```
Comment (cgrayson) [inline src/solar.coffee]: Add a test for this default?

Comment (alexkwolfe) [inline src/solar.coffee]: And null test `vars.velocify`. :)

**Comment by cgrayson** [comment]
Otherwise LGTM.

**Pull request merged**